src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializersTests.Accessibility.cs

```diff
@@ -0,0 +1,392 @@
+        }
+
+        [Fact]
+        public void NestedTypeVisibilityMayBePublic()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    public class Nested
+    {
+        [ModuleInitializer]
+        public static void M() { }
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void NestedTypeVisibilityMayBeInternal()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    internal class Nested
+    {
+        [ModuleInitializer]
+        public static void M() { }
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void NestedTypeVisibilityMayBeProtectedInternal()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    protected internal class Nested
+    {
+        [ModuleInitializer]
+        public static void M() { }
+    }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void MayBeDeclaredByStruct()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+struct S
+{
+    [ModuleInitializer]
+    internal static void M() { }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+            compilation.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void MayBeDeclaredByInterface()
+        {
+            string source = @"
+using System.Runtime.CompilerServices;
+
+interface I
+{
+    [ModuleInitializer]
+    internal static void M() { }
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            var compilation = CreateCompilation(source, parseOptions: s_parseOptions);
+
+            compilation.VerifyDiagnostics(
+                // (7,26): error CS8701: Target runtime doesn't support default interface implementation.
+                //     internal static void M() { }
+                Diagnostic(ErrorCode.ERR_RuntimeDoesNotSupportDefaultInterfaceImplementation, "M").WithLocation(7, 26)
+                );
+        }
+
+        [Fact]
+        public void ImplicitPublicInterfaceMethodVisibility()
```

Comment:
>ImplicitPublicInterfaceMethodVisibility [](start = 20, length = 39)

Should be  a success case.

Reply:
This test demonstrates the success case.

Reply:
> This test demonstrates the success case.

Compiling with an error, even different one, is not a success. Success is to run and observe expected behavior at runtime.

---
In reply to: [418183993](https://github.com/dotnet/roslyn/pull/43301#discussion_r418183993) [](ancestors = 418183993)

Reply:
So my `// Intentionally demonstrated without DIM support.` wasn't such a bright idea. Okay!

Reply:
> So my // Intentionally demonstrated without DIM support. wasn't such a bright idea. Okay!

No. This is a good test, the scenario should be supported,  we should do complete testing. See existing tests for DIM, how they make things runnable, etc. 

---
In reply to: [418201461](https://github.com/dotnet/roslyn/pull/43301#discussion_r418201461) [](ancestors = 418201461)

Reply:
Would `DefaultInterfaceImplementationTests.ValidateMethodImplementation_011` be a good approach to emulate? It looks like it compiles by targeting NetStandardLatest and then skips verification and running if the platform doesn't support DIM.

Reply:
I went with this approach.

Reply:
> Would DefaultInterfaceImplementationTests.ValidateMethodImplementation_011 be a good approach to emulate? It looks like it compiles by targeting NetStandardLatest and then skips verification and running if the platform doesn't support DIM.

Yes.

---
In reply to: [418354429](https://github.com/dotnet/roslyn/pull/43301#discussion_r418354429) [](ancestors = 418354429)